features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/resources/MD5HashGenerator.kt

```diff
@@ -33,7 +32,21 @@ internal class MD5HashGenerator(
         }
     }
 
+    private fun convertToHexString(byteArray: ByteArray): String {
+        @Suppress("UnsafeThirdPartyFunctionCall") // byte array size is always positive.
+        val result = StringBuilder(byteArray.size * 2)
+        for (byte in byteArray) {
+            val intVal = byte.toInt() and BYTE_MASK
+            result.append(HEX_CHARS[intVal ushr HEX_SHIFT]) // Append first half of byte
+            result.append(HEX_CHARS[intVal and LOWER_NIBBLE_MASK]) // Append second half of byte
+        }
+        return result.toString()
+    }
```

Comment:
maybe add a test? the output should match then with "%02x" format output

Reply:
this private function is already covered by `MD5HashGeneratorTest` before, but to make it reusable in the future, I move this to a ext file in `dd-sdk-android-internal` module, and created a dedicated test for it.